# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/blindcoder/acerhk/depmod_2.6.18.patch
# ROCK Linux is Copyright (C) 1998 - 2007 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./Makefile.orig	2007-05-16 21:53:04.000000000 +0200
+++ ./Makefile	2007-05-16 23:06:57.000000000 +0200
@@ -3,9 +3,9 @@
 #KERNELSRC=/usr/src/linux
 KERNELSRC?=/lib/modules/`uname -r`/build
 # Starting with 2.6.18, the kernel version is in utsrelease.h instead of version.h, accomodate both cases
-KERNELVERSION=$(shell awk -F\" '/REL/ {print $$2}' $(shell grep -s -l REL $(KERNELSRC)/include/linux/version.h $(KERNELSRC)/include/linux/utsrelease.h))
+#KERNELVERSION=$(shell awk -F\" '/REL/ {print $$2}' $(shell grep -s -l REL $(KERNELSRC)/include/linux/version.h $(KERNELSRC)/include/linux/utsrelease.h))
+KERNELVERSION:=$(shell if [ -f $(KERNELSRC)/include/linux/utsrelease.h ]; then awk -F\" '/REL/ {print $$2}' $(KERNELSRC)/include/linux/utsrelease.h; else awk -F\" '/REL/ {print $$2}' $(KERNELSRC)/include/linux/version.h; fi)
 KERNELMAJOR=$(shell echo $(KERNELVERSION)|head -c3)
-
 # next line is for kernel 2.6, if you integrate the driver in the kernel tree
 # /usr/src/linux/drivers/acerhk - or something similar
 # don't forget to add the following line to the parent dir's Makefile:
